Ghidra/Features/PDB/src/pdb/cpp/iterate.cpp

```diff
@@ -280,137 +269,135 @@ void dumpFunctionStackVariables( DWORD rva )
 	};
 }
 
-void dumpFunctionLines( IDiaSymbol* pSymbol, IDiaSession* pSession )
+void dumpFunctionLines( IDiaSymbol& pSymbol, IDiaSession& pSession )
 {
 	ULONGLONG length = 0;
 	DWORD isect = 0;
 	DWORD offset = 0;
-	pSymbol->get_addressSection( &isect );
-	pSymbol->get_addressOffset( &offset );
-	pSymbol->get_length( &length );
+	pSymbol.get_addressSection( &isect );
+	pSymbol.get_addressOffset( &offset );
+	pSymbol.get_length( &length );
 	if ( isect == 0 || length <= 0 ) {
 		return;
 	}
-	IDiaEnumLineNumbers * pLines;
-	if (pSession->findLinesByAddr( isect, offset, static_cast<DWORD>( length ), &pLines ) < 0) {
+
+    CComPtr<IDiaEnumLineNumbers> pLines;
+	if (FAILED(pSession.findLinesByAddr( isect, offset, static_cast<DWORD>( length ), &pLines ))) {
 		return;
 	}
-	IDiaLineNumber * pLine;
-	DWORD celt;
+
+	DWORD celt = 0;
 	while ( 1 ) {
-		if (pLines->Next( 1, &pLine, &celt ) < 0) {
+		CComPtr<IDiaLineNumber> pLine;
+		if (FAILED(pLines->Next( 1, &pLine, &celt ))) {
 			break;
 		}
 		if (celt != 1) {
 			break;
 		}
 
-		IDiaSymbol * pComp;
+        CComPtr<IDiaSymbol> pComp;
 		pLine->get_compiland( &pComp );
 
-		IDiaSourceFile * pSrc;
+        CComPtr<IDiaSourceFile> pSrc;
 		pLine->get_sourceFile( &pSrc );
 
-		BSTR sourceFileName = NULL;
-		pSrc->get_fileName( &sourceFileName );
+		BSTR temp = NULL;
+		pSrc->get_fileName(&temp);
+		bstr_t sourceFileName;
+		sourceFileName.Attach(temp);
 
-		DWORD addr;
+		DWORD addr = 0;
 		pLine->get_relativeVirtualAddress( &addr );
 
-		DWORD start;
+		DWORD start = 0;
 		pLine->get_lineNumber( &start );
-		DWORD end;
+		DWORD end = 0;
 		pLine->get_lineNumberEnd( &end );
 
-		printf("%s<line_number source_file=\"%ws\" start=\"0x%x\" end=\"0x%x\" addr=\"0x%x\" /> \n", 
-					indent(12), sourceFileName, start, end, addr);
-
-		pLine = NULL;
+		printf("%S<line_number source_file=\"%S\" start=\"0x%x\" end=\"0x%x\" addr=\"0x%x\" /> \n", 
```

Comment:
is %S correct for sourceFileName.GetBSTR()?  I thought you said %ws, but not sure.